docker/archive/multi_source.go

```diff
@@ -0,0 +1,68 @@
+package archive
+
+import (
+	"github.com/containers/image/v5/docker/internal/tarfile"
+	"github.com/containers/image/v5/docker/reference"
+	"github.com/containers/image/v5/types"
+	"github.com/pkg/errors"
+)
+
+// MultiImageSource manages a single Docker archive, allows listing its contents
+// and accessing individual images with less overhead than creating image references
+// individually (because the archive is, if necessary, copied or decompressed only once).
+type MultiImageSource struct {
```

Comment:
I’m not entirely happy about this name - it’s not an `ImageSource`. `docker/archive.Reader`? `MultiReader`? `MultiImageReader`?

Reply:
Trying `docker/archive.Reader` — it’s not just for multi-image readers, a caller that wants to open a single-image archive, extract the tag from the manifest, and read the single image, would benefit as well.